src/Types/OnChainData.cs

```diff
@@ -0,0 +1,144 @@
+using System.Formats.Cbor;
+
+using Concordium.Sdk.Helpers;
+
+namespace Concordium.Sdk.Types;
+
+/// <summary>
+/// Represents data to be stored or that was stored on-chain as part of a transaction.
+///
+/// This can be any data which is at most <see cref="MaxLength"/> bytes, but the convention is
+/// to use CBOR encoded data.
+/// </summary>
+public record OnChainData : IEquatable<OnChainData>
```

Comment:
I would expect this to be a class due to it’s methods like `TryCborDecodeToString`.